polygeist/tools/cgeist/driver.cc

```diff
@@ -1164,6 +1165,58 @@ template <typename T> static void filterFunctions(T Module) {
     Func.erase();
 }
 
+static bool readHostModule(MLIRContext &context, ModuleOp module,
+                           llvm::StringRef inputFilename) {
+  Block &dest = module.getBodyRegion().front();
+  assert(dest.getOperations().size() == 1 &&
+         "Expecting a single operation in the original module");
+
+  // Set up the input file.
+  std::string errorMessage;
+  auto file = openInputFile(inputFilename, &errorMessage);
```

Comment:
is it useful to check the suffix of the filename to be `.mlir`?

Reply:
I don't think we should enforce that. In the end, "extensions" mean nothing. Also, I don't think other tools do. I'll change the commit message to avoid confusion.